source/mqtt_agent.c

```diff
@@ -814,11 +845,23 @@ static void clearPendingAcknowledgments( MQTTAgentContext_t * pMqttAgentContext
     {
         if( pendingAcks[ i ].packetId != MQTT_PACKET_ID_INVALID )
         {
-            /* Receive failed to indicate network error. */
-            concludeCommand( pMqttAgentContext, pendingAcks[ i ].pOriginalCommand, MQTTRecvFailed, NULL );
+            bool clearEntry = true;
+
+            if( clearOnlySubUnsubEntries &&
```

Comment:
Nit: Perhaps we should `assert( pendingAcks[ i ].pOriginalCommand != NULL );` before dereferencing it.